test/components/agents/test_agent.py

```diff
@@ -254,3 +254,57 @@ def test_chat_generator_must_support_tools(self, weather_tool):
 
         with pytest.raises(TypeError, match="MockChatGeneratorWithoutTools does not accept tools"):
             Agent(chat_generator=chat_generator, tools=[weather_tool])
+
+    def test_multiple_llm_responses_with_tool_call(self, monkeypatch, weather_tool):
+        monkeypatch.setenv("FAKE_OPENAI_KEY", "fake-key")
+        generator = OpenAIChatGenerator(api_key=Secret.from_env_var("FAKE_OPENAI_KEY"))
+
+        mock_messages = [
+            ChatMessage.from_assistant("First response"),
+            ChatMessage.from_assistant(
+                tool_calls=[ToolCall(tool_name="weather_tool", arguments={"location": "Berlin"})]
+            ),
+        ]
+
+        agent = Agent(chat_generator=generator, tools=[weather_tool], max_runs_per_component=1)
+        agent.warm_up()
+
+        # Patch agent.chat_generator.run to return mock_messages
+        agent.chat_generator.run = MagicMock(return_value={"replies": mock_messages})
+
+        result = agent.run([ChatMessage.from_user("Hello")])
+
+        assert "messages" in result
+        assert len(result["messages"]) == 4
+        assert (
+            result["messages"][-1].tool_call_result.result
+            == "{'weather': 'mostly sunny', 'temperature': 7, 'unit': 'celsius'}"
+        )
+
+    def test_exit_conditions_checked_across_all_llm_messages(self, monkeypatch, weather_tool):
+        monkeypatch.setenv("FAKE_OPENAI_KEY", "fake-key")
```

Comment:
If we use `monkeypatch.setenv("OPENAI_API_KEY", "fake-key")` here then we can simplify the next line to 
`generator = OpenAIChatGenerator()`